### ISSUE TITLE
[no-uncalled-signals] Wrongly reports called signal in expressions

### ISSUE THREAD

**Issue by muuvmuuv** [angular-eslint/angular-eslint#2490]
**Description and reproduction of the issue**

This looks like wrongly reported, the Signal is called just not assigned to a variable before (which temp fixes this issue).

YOUR LINK HERE: [...](https://stackblitz.com/edit/angular-eslint-repros-kyw2supk?file=src%2Fmain.ts)

**Versions**

| package                         | version |
| ------------------------------- | ------- |
| `@angular-eslint/eslint-plugin` | `^19.6.0` |
| `@typescript-eslint/parser`     | `X.Y.Z` |
| `ESLint`                        | `^9.27.0` |
| `node`                          | `22.x` |

```sh
Angular CLI: 20.0.0
Node: 22.16.0
Package Manager: pnpm 10.11.0
OS: darwin arm64

Angular: 20.0.0
... animations, build, cdk, cli, common, compiler, compiler-cli
... core, forms, material, platform-browser
... platform-browser-dynamic, router, service-worker

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.2000.0
@angular-devkit/build-angular   20.0.0
@angular-devkit/core            20.0.0
@angular-devkit/schematics      20.0.0
@schematics/angular             20.0.0
rxjs                            7.8.2
typescript                      5.8.2
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by JamesHenry** [comment]
@claude This is a summary of the bug in the `no-uncalled-signals` rule.

This incorrectly reports a lint violation

```ts
export class AppComponent {
  readonly test = signal<boolean>(false);

  constructor() {
    effect(() => {
      if (this.test()) {                 // <------------- violation reported here
        console.log('Hey');
      } else {
        console.log('Hoo');
      }
    });
  }
}
```

This correctly does not report a lint violation:

```ts
export class AppComponent {
  readonly test = signal<boolean>(false);

  constructor() {
    const t = this.test;
    effect(() => {
      if (t()) {
        console.log('Hey');
      } else {
        console.log('Hoo');
      }
    });
  }
}
```

As you can see, the work around is to set an interim variable, but this should not be necessary.

Fix the logic of the rule and add test coverage for this in `packages/eslint-plugin/tests/rules/no-uncalled-signals/cases.ts`. You can run the tests for just this rule with:

```sh
pnpm nx test eslint-plugin packages/eslint-plugin/tests/rules/no-uncalled-signals/spec.ts
```

**Comment by muuvmuuv** [comment]
No f* way Claude just fixed that, I am impressed haha 😂

Does this also check for inline conditional or optional operators?

```ts
this.#control = this.__formControl() ?? this.getControl(this.name())
```
